core/src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -243,61 +309,59 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
     configConvert: Lazy[ConfigConvert[T]],
     cbf: CanBuildFrom[F[T], T, F[T]]) = new ConfigConvert[F[T]] {
 
-    override def from(config: ConfigValue): Try[F[T]] = {
+    override def from(config: ConfigValue): ConfigReaderResult[F[T]] = {
       config match {
         case co: ConfigList =>
-          traverseTry(co.asScala, cbf())(configConvert.value.from)
+          val z: Either[ConfigReaderFailures, Builder[T, F[T]]] = Right(cbf())
+
+          // we called all the failures in the list
+          co.asScala.foldLeft(z) {
+            case (acc, value) =>
+              combineResults(acc, configConvert.value.from(value))(_ += _)
+          }.right.map(_.result())
         case o: ConfigObject =>
-          for {
-            indexedEntries <- traverseTry(o.asScala, new collection.mutable.ListBuffer[(Int, ConfigValue)]) {
-              case (keyString, value) => Try(keyString.toInt -> value).recoverWith(reportBadIndex(keyString))
-            }
-            sortedValues = indexedEntries.sortBy(_._1).map(_._2)
-            result <- traverseTry(sortedValues, cbf())(configConvert.value.from)
-          } yield result
+          val z: Either[ConfigReaderFailures, List[(Int, T)]] = Right(List.empty[(Int, T)])
+          def keyValueReader(key: String, value: ConfigValue): ConfigReaderResult[(Int, T)] = {
+            val keyResult = catchReadError(key, _.toInt).left.flatMap(t => fail(CannotConvert(key, "Int",
```

Comment:
We should consider using a different failure here other than `CannotConvert`. If we do, we can provide more contextual information by concatenating the remaining path to the `key` through the `improveFailure` method.